language/move-lang/src/expansion/byte_string.rs

```diff
@@ -0,0 +1,140 @@
+    if !context.has_errors() {
+        Ok(buffer)
+    } else {
+        Err(context.get_errors())
+    }
+}
+
+fn decode_(context: &mut Context, buffer: &mut Vec<u8>, chars: Vec<char>) {
+    let len = chars.len();
+    let mut i = 0;
+    macro_rules! next_char {
+        () => {{
+            let c = chars[i];
+            i += 1;
+            c
+        }};
+    }
+    macro_rules! next_char_opt {
+        () => {{
+            if i < len {
+                Some(next_char!())
+            } else {
+                None
+            }
+        }};
+    }
+    while i < len {
+        let cur = i;
+        let c = next_char!();
+        if c != '\\' {
+            push(buffer, c);
+            continue;
+        }
+
+        match next_char!() {
+            'n' => push(buffer, '\n'),
+            'r' => push(buffer, '\r'),
+            't' => push(buffer, '\t'),
+            '\\' => push(buffer, '\\'),
+            '0' => push(buffer, '\0'),
+            '"' => push(buffer, '"'),
+            'x' => {
+                let d0_opt = next_char_opt!();
+                let d1_opt = next_char_opt!();
+                let hex = match (d0_opt, d1_opt) {
+                    (Some(d0), Some(d1)) => {
+                        let mut hex = String::new();
+                        hex.push(d0);
+                        hex.push(d1);
+                        hex
+                    }
+
+                    // Unexpected end of text
+                    (d0_opt @ Some(_), None) | (d0_opt @ None, None) => {
+                        let h = match d0_opt {
+                            Some(d0) => format!("{}", d0),
+                            None => "".to_string(),
+                        };
+                        let err_text = format!(
+                            "Invalid escape: '\\x{}'. Hex literals are represented by two symbols: \
+                            [\\x00-\\xFF].",
+                            h
+                        );
+                        context.error(cur, len, err_text);
+                        return;
+                    }
+
+                    // There was a second digit but no first?
+                    (None, Some(_)) => unreachable!(),
+                };
+                match hex::decode(hex) {
+                    Ok(hex_buffer) => buffer.extend(hex_buffer),
+                    Err(hex::FromHexError::InvalidHexCharacter { c, index }) => {
+                        let err_text = format!("Invalid hexadecimal character: '{}'", c);
+                        context.error(cur + 2 + index, cur + 2 + index, err_text);
+                    }
+                    Err(_) => unreachable!("ICE unexpected error parsing hex byte string value"),
+                }
+            }
+            c => {
+                context.error(cur, cur + 2, format!("Invalid escape sequence: '\\{}'", c));
+            }
+        }
+    }
+}
+
+fn push(buffer: &mut Vec<u8>, ch: char) {
+    let mut next = vec![0; ch.len_utf8()];
```

Comment:
Since we decided not to support Unicode escapes in byte strings, these are purely ASCII strings and there is no need to encode the bytes to UTF-8. (And, we should not do that since it is confusing.)

Reply:
Agreed. But it was the only I could figure out to convert it into u8. 

Let me play around with a few other things. If I can't get it to work, I'll leave it as is

Reply:
Oops, turns I can just cast it to `u8`. Finding out what things cast into what things is Rust documentation is a little tricky....